File: Codeforces/Div2/357/C/Main.java
import java.util.Scanner;
import java.util.PriorityQueue;
import java.util.List;
import java.util.ArrayList;

public class Main {
  static int n;
  static List< String > ans;

  public static void main(String[] args){
    Scanner sc = new Scanner(System.in);
    ans = new ArrayList< String >();

    n = sc.nextInt();

    PriorityQueue< Integer > pq = new PriorityQueue< Integer >();
    for(int i = 0; i < n; i++){
      int opNum = 0;
      String op = sc.next();
      if("insert".equals(op) || "getMin".equals(op)){
        opNum = sc.nextInt();
      }
      if("insert".equals(op)){
        pq.offer(opNum);
        ans.add(op + " " + opNum);
      }
      if("removeMin".equals(op)){
        if(pq.isEmpty()){
          ans.add("insert 1");
          ans.add("removeMin");
        }
        else{
          pq.poll();
          ans.add("removeMin");
        }
      }

      if("getMin".equals(op)){
        if(!pq.contains(opNum)){
          pq.offer(opNum);
          ans.add("insert " + opNum);
        }
        while(pq.peek() != opNum){
          pq.poll();
          ans.add("removeMin");
        }
        ans.add("getMin " + opNum);
      }
    }

    System.out.println(ans.size());
    for(int i = 0; i < ans.size(); i++){
      System.out.println(ans.get(i));
    }
  }
}

File: Codeforces/Div2/357/C/main.cpp
#include <cstdio>
#include <iostream>
#include <vector>
#include <string>
#include <queue>
#include <set>
using namespace std;

int n;
vector< string > ans;
priority_queue< int, vector< int >, greater< int > > pq;
multiset< int > s;

int main(){
  scanf("%d", &n);

  for(int i = 0; i < n; i++){
    int opNum = 0;
    char tmp[128];
    scanf("%s", tmp);
    string op(tmp);
    if(op.compare("insert") == 0 || op.compare("getMin") == 0){
      scanf("%d", &opNum);
    }
    if(op.compare("insert") == 0){
      pq.push(opNum);
      s.insert(opNum);
      ans.push_back(op + " " + to_string(opNum));
    }
    if(op.compare("removeMin") == 0){
      if(pq.empty()){
        ans.push_back("insert 1");
        ans.push_back("removeMin");
      }
      else{
        s.erase(pq.top());
        pq.pop();
        ans.push_back("removeMin");
      }
    }

    if(op.compare("getMin") == 0){
      if(s.count(opNum) == 0){
        pq.push(opNum);
        s.insert(opNum);
        ans.push_back("insert " + to_string(opNum));
      }
      while(pq.top() != opNum){
        s.erase(pq.top());
        pq.pop();
        ans.push_back("removeMin");
      }
      ans.push_back("getMin " + to_string(opNum));
    }
  }

  cout << ans.size() << endl;
  for(int i = 0; i < ans.size(); i++){
    printf("%s\n", ans[i].c_str());
  }

  return 0;
}
